LuckySpin/Services/SpinService.cs

```diff
@@ -12,14 +13,25 @@ public class SpinService : ISpinService //Makes this class extend the Interface
         public SpinService(ISpinRepository sr)
         {
             spinRepository = sr;
-
         }
 
 
-        public double CalculateAvgWins()
+        public double CalculateAvgWins(bool isWinning)
         {
             //TODO: Write logic to use the "real" spinRepository NOT the test data
-            return .1; 
+            double winningCount = 0;
+            double spinCounts = spinRepository.GetCount() + 1;
+
+            if (isWinning)
+                winningCount = 1;
+
+            foreach (Spin spin in spinRepository.GetSpins())
+            {
+                if (spin.IsWinning)
+                    winningCount++;
+            }
```

Comment:
This works - good job. Consider refactoring using the Count() method from Lambda Extension methods